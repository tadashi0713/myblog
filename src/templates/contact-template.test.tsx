import React from "react";
import renderer from "react-test-renderer";
import ContactTemplate from "./contact-template";

describe("ContactTemplate", () => {
  const props = {
    data: {
      site: {
        siteMetadata: {
          title: "test",
          subtitle: "test",
        },
      },
    },
  };

  it("renders correctly", () => {
    const tree = renderer.create(<ContactTemplate {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
