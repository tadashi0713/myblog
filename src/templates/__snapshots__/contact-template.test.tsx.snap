// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactTemplate renders correctly 1`] = `
<div
  className="layout"
>
  <link
    crossOrigin="anonymous"
    href="https://maxcdn.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css"
    integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS"
    rel="stylesheet"
  />
  <StaticQuery
    render={[Function]}
  />
  <div
    className="page"
  >
    <div
      className="page__inner"
    >
      <h1
        className="page__title"
      >
        Contact / お問い合わせ
      </h1>
      <div
        className="page__body"
      >
        <form
          action="/contact/thanks/"
          className=""
          data-netlify="true"
          data-netlify-honeypot="bot-field"
          data-netlify-recaptcha="true"
          method="post"
          name="contact"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
            hidden={true}
          >
            <label
              className="form-label"
            >
              Don’t fill this out: 
            </label>
            <input
              className="form-control"
              name="bot-field"
              onChange={[Function]}
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Your name / お名前
            </label>
            <input
              className="form-control"
              name="name"
              onChange={[Function]}
              required={true}
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Your Email / メールアドレス
            </label>
            <input
              className="form-control"
              name="email"
              onChange={[Function]}
              required={true}
              type="email"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Message / お問い合わせ内容
            </label>
            <textarea
              className="form-control"
              name="message"
              onChange={[Function]}
              required={true}
              rows="5"
            />
          </div>
          <div />
          <br />
          <button
            className="btn btn-warning"
            disabled={false}
            type="submit"
          >
            Send
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
